code/missionui/missionweaponchoice.cpp

```diff
@@ -855,7 +855,7 @@ void draw_3d_overhead_view(int model_num,
 			if ((weapon_array[x] == selected_weapon_class && hovered_weapon_slot < 0) ||
 				x == hovered_weapon_slot) {
 				Assert(num_found < NUM_ICON_FRAMES);
```

Comment:
This would need to reflect the now smaller array size of only 4 instead of 6 and check against MAX_SHIP_SECONDARY_BANKS.

Reply:
Good catch